src/components/Payroll/PayrollConfiguration/PayrollConfiguration.tsx

```diff
@@ -81,7 +91,20 @@ export const Root = ({ onEvent, companyId, payrollId, dictionary }: PayrollConfi
     onEvent(event as EventType, payload)
   }
 
-  if (isPreparePayrollPending) {
+  const isCalculating =
+    payrollData.payrollShow?.processingRequest?.status ===
+    PayrollProcessingRequestStatus.Calculating
+  const isCalculated =
+    payrollData.payrollShow?.processingRequest?.status ===
+    PayrollProcessingRequestStatus.CalculateSuccess
+
+  useEffect(() => {
+    if (isCalculated) {
+      onEvent(componentEvents.RUN_PAYROLL_CALCULATED)
+    }
+  }, [isCalculated, onEvent])
+
+  if (isPreparePayrollPending || isCalculating) {
```

Comment:
Do we have a follow up ticket to display the loading UX aaron had in the design for when this is calculating? Do we also need to disable the calculate button in PayrollConfigurationPresentation while this is calculating?

Reply:
`const { data: payrollData, refetch } = usePayrollsGetSuspense(`
afaik I am in fact using a query hook that should suspend...